Adding litecoin and litecoin testnet support

kaykurokawa opened the pull request:
Adding litecoin and litecoin testnet support

kaykurokawa committed:
adding litecoin and litecoin testnet support

kanzure commented on altcoin/core/__init__.py:
```diff
@@ -216,6 +216,17 @@ class CoreDogeTestNetParams(CoreDogeMainParams):
     NAME = 'dogecoin_test'
     GENESIS_BLOCK = CAltcoinBlock.deserialize(x('010000000000000000000000000000000000000000000000000000000000000000000000696ad20e2dd4365c7459b4a4a5af743d5e92c6da3229e6532cd605f6533f2a5bb9a7f052f0ff0f1ef7390f000101000000010000000000000000000000000000000000000000000000000000000000000000ffffffff1004ffff001d0104084e696e746f6e646fffffffff010058850c020000004341040184710fa689ad5023690c80f3a49c8f13f8d45b ...
 
+class CoreLtcMainParams(bitcoin.core.CoreChainParams):
```
Would be nice if these are linked to where in the Litecoin source code they are pulled. (in comments in the source code)

rnicoll commented:
Looks good by eye, want to test it properly before I merge, may be this weekend, but will try for before!

kaykurokawa committed:
Correcting DOGE and LTC proof of work limit. Adding comment links to litecoin/dogecoin source code for finding various parameters

kaykurokawa commented:
Hello, I added comment links to the litecoin source code as requested. 
I also corrected proof of work limit on dogecoin /litecoin , they were 2^256-1 >> 32 when it should be 2^256-1 >> 20  

https://github.com/litecoin-project/litecoin/blob/master-0.8/src/main.cpp#L39
https://github.com/dogecoin/dogecoin/blob/master/src/chainparams.cpp#L115

rnicoll commented:
Good catch with the proof of work limit. Trying to figure out a good test protocol now.
